veps/sig-compute/vgpu_live-migration.md

```diff
@@ -0,0 +1,97 @@
+# VEP #109: Implement vGPU Enabled Live Migration
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [X] (R) Enhancement issue created, which links to VEP dir in [kubevirt/enhancements] (not the initial VEP PR)
+- [] (R) Target version is explicitly mentioned and approved
+- [] (R) Graduation criteria filled
+
+## Overview
+
+This is a proposal to allow live migrations in KubeVirt to work for VMs with a single vGPU, exposed by mdev, between two nodes in the same cluster with identical GPUs.
+
+## Motivation
+
+GPU usage is increasing with more and more companies running AI workloads, so companies are now requesting live migration to support GPU enabled VMs.
+
+## Goals
+
+* Address a common live migration problem where the target needs to update the destination Libvirt XML. In the case of mdevs, it needs to update the mdev UUID in the XML.
+* Support single vGPU enabled live migrations for both nodes that are using the Nvidia GPU Operator and clusters that are using KubeVirt’s generic device plugin for mdev.
+* Support single vGPU enabled live migrations with minimal data lost due to high dirty rates.
+
+## Non Goals
+
+* Do not want to change the live migration workflow for non vGPU enabled VMs.
+* Do not support live migration for passthrough or SRIOV vGPU
+* Do not support cross-cluster live migrations
+* Do not support live migrations for VMs with multiple vGPUs
+
+## Definition of Users
+
+**KubeVirt Administrators:** Users who have cluster wide privileges to trigger APIs to manage a cluster.
+**KubeVirt Owner:** VM workload owners who want high availability for their VMs.
+
+## User Stories
+
+As a KubeVirt admin/owner, I want to be able to live migrate my VMs that have a vGPU.
+
+## Repos
+
+https://github.com/kubevirt/kubevirt
+
+## Design
+
+In order to live migrate a vGPU, we must first check that the host-driver versions for the GPU are the same on each node for compatability. Then, the destination XML needs to be updated with the mdev UUID that KubeVirt allocates on the target virt-launcher. To do this, we use a QEMU hook that interacts with the target `virt-launcher` pod. Live migrations have multiple phases, and the QEMU hook is called at each phase. The hook will have conditional statements to only execute at the "prepare begin" phase. Hooks receive the phase and XML from stdin, and the updated XML is sent to stdout.
+
+Once the destination XML contains the correct fields, the live migration can begin. Libvirt/QEMU already support vGPU live migration for mdev (since Libvirt 8.6.0 and QEMU 8.1.0) and will do the actual migration, so no further work is needed by KubeVirt to migrate the vGPU. Some migration configs at the Libvirt/QEMU level, such as the migration method or downtime limit, may be necessary however.
```

Comment:
Can you please add an xml snippet of the information that would need to change in the domain?
Similar to what we have in https://github.com/kubevirt/enhancements/blob/main/veps/sig-network/ordinal-naming-scheme-upgrade-path.md?plain=1#L100 .